### PR TITLE
Migrate to SqlDelight

### LINKED ISSUE #1088: Migrate from Room to SQLDelight

### PR DESCRIPTION
**Pull request by chrisbanes**
This is a very manual and mechanical migration. Also very difficult to know if it's working until it's all complete.

- [x] Add skeleton setup in new module
- [x] `EpisodeDao` implementation
- [x] `EpisodeWatchEntryDao` implementation
- [x] `FollowedShowsDao` implementation
- [x] `LastRequestDao` implementation
- [x] `LibraryShowsDao` implementation
- [x] `PopularDao` implementation
- [x] `RecommendedDao` implementation
- [x] `RelatedShowsDao` implementation
- [x] `SeasonsDao` implementation
- [x] `ShowFtsDao` implementation. Will need to figure out FTS support too?
- [x] `ShowTmdbImagesDao` implementation
- [x] `TiviShowDao` implementation
- [x] `TrendingDao` implementation
- [x] `UserDao` implementation
- [x] `WatchedShowDao` implementation
- [x] Re-enable `allWarningsAsErrors`
- [x] Add last few DB version migrations?

Will eventually close #1088

### CODE REVIEW

File: data/db-sqldelight/src/commonMain/sqldelight/migrations/31.sqm
Comment (JakeWharton) [inline data/db-sqldelight/src/commonMain/sqldelight/migrations/31.sqm:1]: If you're not copying migrations for every old version then this is the only file you should need. You don't need empty ones for previous versions.

File: data/db-sqldelight/src/commonMain/sqldelight/migrations/.sqm
Comment (JakeWharton) [inline data/db-sqldelight/src/commonMain/sqldelight/migrations/.sqm:1]: Bash looping error?

File: data/db-sqldelight/src/commonMain/sqldelight/migrations/31.sqm
Comment (chrisbanes) [reply data/db-sqldelight/src/commonMain/sqldelight/migrations/31.sqm:1]: TIL. I will probably end up migrating the last few DB migration code, but removing for now and adding a TODO

File: data/db-sqldelight/src/commonMain/sqldelight/migrations/.sqm
Comment (chrisbanes) [reply data/db-sqldelight/src/commonMain/sqldelight/migrations/.sqm:1]: Indeed 🤦

File: room2sqld.py
Comment (AlecStrong) [inline room2sqld.py:1]: neat

**Comment by chrisbanes** [comment]
IT'S FINALLY READY 😪

**Pull request merged**